plugins-structure/README.md

```diff
@@ -0,0 +1,579 @@
+4. [Core](#core)
+
+## Requirements
+
+The following guide will assume you are familiar with the tools listed below. We
+recommend you to take some time to study them before proceeding.
+
+- [JavaScript](https://developer.mozilla.org/docs/Web/JavaScript)
+- [Micro-Frontends](https://micro-frontends.org/)
+- [Lerna](https://lerna.js.org/)
+- [Yarn Workspaces](https://yarnpkg.com/features/workspaces)
+- [Politeia](https://github.com/decred/politeia)
+- [Redux Toolkit](https://redux-toolkit.js.org/)
+- [Window History](https://developer.mozilla.org/docs/Web/API/Window/history)
+
+## Introduction
+
+Politeia is the underlying platform that provides a versioned, timestamped data
+store [[2]](#2), which can be used for creating new applications on top of it.
+
+> The original motivations behind Politeia were to create a platform that would
+> allow the Decred stakeholders to manage the Decred treasury. This platform
+> would let individuals and teams submit proposals (i.e. requests for funding
+> from the treasury) and would facilitate the Decred stakeholders voting on
+> whether to fund the proposal or reject it. The data would be stored off-chain,
+> but would be episodically timestamped onto the Decred blockchain" - _Politeia:
+> Pi 2021 Q3 (Proposal System)_ [[2]](#2).
+
+However, the stored data could be any set of data, not exclusively proposals,
+which could also be used as a more generic storage, such as contractors
+payments management, proposals discussions, etc.
+
+The current design would allow developers to create new plugins to interact,
+extend and manage the Record functionality. Therefore, we can use Politeia
+[[1]](#1) architecture to create more applications, such as
+[Decred's Proposal system (Pi)](https://proposals.decred.org) and
+[Decred's Contractor Management System (CMS)](https://cms.decred.org/)
+_(CMS not implemented on the new architecture yet)_.
+
+PoliteiaGUI is the Guest User Interface side of Politeia, which stands for the
+same principles of a **plugin-composable architecture**. It is responsible for
+displaying the multiple stored Records on a user-friendly interface, helping
+people who aren't familiar with programming and CLI management also interact
+and use the functionalities provided by the regarded system.
+
+The architecture consists in a composition of one or more plugins that extend
+the Core functionalities. For example, plugins can enhance Records by providing
+extra information, data, statuses, etc. using its own isolated design. This will
+provide a reusable structure for multiple applications, which will provide a
+better code maintainance and scalability.
+
+On the Frontend side, it means we can provide different UXs depending on the
+plugins combinations. We can integrate multiple plugins for customizing some
+app, and still not trigger collateral effects on other plugins.
+
+## GUI Composable Plugin Architecture
+
+The GUI architecture is a mono-repository structure, composed by
```

Comment:
I don't think we could define our structure as micro-frontends. We use a lot of the same concepts but micro-frontends have independent deploy pipelines and micro-apps are combined on run-time. I'd rather we don't call our `plugins` micro-frontends and just stick with plugins.